public/app/core/components/AppChrome/AppChromeSidePane.tsx

```diff
@@ -0,0 +1,24 @@
+import { useLayoutEffect } from 'react';
+import * as React from 'react';
+
+import { useGrafana } from 'app/core/context/GrafanaContext';
+
+export interface AppChromeUpdateProps {
+  left?: React.ReactNode;
+}
+/**
+ * This needs to be moved to @grafana/ui or runtime.
+ * This is the way core pages and plugins update the breadcrumbs and page toolbar actions
+ */
+export const AppChromeSidePane = React.memo<AppChromeUpdateProps>(({ left }: AppChromeUpdateProps) => {
+  const { chrome } = useGrafana();
+
+  // We use useLayoutEffect here to make sure that the chrome is updated before the page is rendered
+  // This prevents flickering actions when going from one dashboard to another for example
+  useLayoutEffect(() => {
+    chrome.update({ sidePaneLeft: left });
+  });
+  return null;
+});
+
+AppChromeSidePane.displayName = 'AppChromeSidePane';
```

Comment:
can we just use `<AppChromeUpdate sidePaneLeft={<Component />} />` directly, or do you think having this separate wrapper is cleaner? 🤔 (i could go either way tbh... right now it feels like a bit of extra boilerplate, but could be useful in the future)

Reply:
@ashharrison90 we could, I tried it but they override each other the way they currently works